edit-and-sync-features-with-feature-service/src/main/java/com/esri/arcgismaps/sample/editandsyncfeatureswithfeatureservice/components/MapViewModel.kt

```diff
@@ -0,0 +1,465 @@
+                FeatureLayer.createWithFeatureTable(featureTable)
+            }
+        }
+
+        // Update the sync button text to show a sync action
+        actionButtonText = getApplication<Application>().getString(R.string.sync_button_text)
+
+        // Update the geodatabase edit state to indicate its ready for edits and syncs
+        geodatabaseEditState = GeodatabaseEditState.READY
+
+    }
+
+    /**
+     * Syncs changes made on either the local [Geodatabase] or web service geodatabase with each other
+     */
+    private fun syncGeodatabase() {
+        // Create parameters for the geodatabase sync task
+        val syncGeodatabaseParameters = SyncGeodatabaseParameters().apply {
+            geodatabaseSyncDirection = SyncDirection.Bidirectional
+            shouldRollbackOnFailure = false
+        }
+
+        geodatabase?.let { geodatabase ->
+            // Set synchronization option for each layer in the geodatabase we want to synchronize
+            syncGeodatabaseParameters.layerOptions +=
+                geodatabase.featureTables.map { featureTable ->
+                    // Create a new sync layer option with the layer id of the feature table
+                    SyncLayerOption(featureTable.serviceLayerId)
+                }
+
+            with(viewModelScope) {
+                // Create the SyncGeodatabaseJob using the parameters and the geodatabase
+                launch(Dispatchers.IO) {
+                    syncGeodatabaseJob = geodatabaseSyncTask.createSyncGeodatabaseJob(
+                        parameters = syncGeodatabaseParameters, geodatabase = geodatabase
+                    )
+
+                    syncGeodatabaseJob?.let { syncGeodatabaseJob ->
+
+                        // Show the dialog
+                        showSyncGeodatabaseJobProgressDialog = true
+
+                        // Create a flow-collection for the job's progress
+                        launch(Dispatchers.Main) {
+                            syncGeodatabaseJob.progress.collect { progress ->
+                                // Display the current job's progress value
+                                syncGeodatabaseJobProgress = progress
+                            }
+                        }
+                        // Start the job and wait for Job result
+                        syncGeodatabaseJob.start()
+                        syncGeodatabaseJob.result().onSuccess {
+                            showSuccessMessage("Sync Complete")
+                        }
+                            .onFailure { messageDialogVM.showMessageDialog("Database did not sync correctly") }
+                    }
+
+                    // Dismiss the dialog
+                    showSyncGeodatabaseJobProgressDialog = false
+
+                    // Set the edit state to indicate geodatabase is ready for edits
+                    geodatabaseEditState = GeodatabaseEditState.READY
+                }
+            }
```

Comment:
In this code block (`with(viewModelScope)`) you launch a `launch(Dispatchers.IO)` and then it seems like inside this coroutine scope you launch a `launch(Dispatchers.Main)` to collect job's progress. Nested usage of different dispatchers would not run the coroutines as you have defined. 

I would suggest to remove the `createSyncGeodatabaseJob` and the setting of the `showSyncGeodatabaseJobProgressDialog` and `syncGeodatabaseJob.start()` to be placed outside outside the coroutine scope. As none of these are suspended calls and don't need to be defined in a launched scope. And only use suspend function calls within your launched coroutines: 
- launch(Dispatchers.Main) to collect job's progress
- launch(Dispatchers.IO) to await the job's result 

```suggestion
            syncGeodatabaseJob = geodatabaseSyncTask.createSyncGeodatabaseJob(
                parameters = syncGeodatabaseParameters,
                geodatabase = geodatabase
            )

            syncGeodatabaseJob?.let { syncGeodatabaseJob ->

                // Show the dialog
                showSyncGeodatabaseJobProgressDialog = true

                // Start the job and wait for Job result
                syncGeodatabaseJob.start()

                with(viewModelScope) {
                    // Create the SyncGeodatabaseJob using the parameters and the geodatabase
                    launch(Dispatchers.IO) {
                        syncGeodatabaseJob.result().onSuccess {
                            showSuccessMessage("Sync Complete")
                        }.onFailure {
                            messageDialogVM.showMessageDialog("Database did not sync correctly")
                        }

                        // Dismiss the dialog
                        showSyncGeodatabaseJobProgressDialog = false

                        // Set the edit state to indicate geodatabase is ready for edits
                        geodatabaseEditState = GeodatabaseEditState.READY
                    }

                    // Create a flow-collection for the job's progress
                    launch(Dispatchers.Main) {
                        syncGeodatabaseJob.progress.collect { progress ->
                            // Display the current job's progress value
                            syncGeodatabaseJobProgress = progress
                        }
                    }
                }
            }
```